### PR TITLE
Fix SparseArray validity logic and give DateTimeParts unique code

### PR DESCRIPTION
**Pull request by robert3005**

### CODE REVIEW

File: vortex-array/src/array/sparse/mod.rs [hunk 1]
```diff
@@ -157,10 +164,10 @@ impl ArrayValidity for SparseArray {
             // of true, and patch values of false.
             Self::try_new_with_offset(
                 self.indices(),
-                ConstantArray::new(false, self.len()).into_array(),
+                ConstantArray::new(true, self.indices().len()).into_array(),
```
Comment (robert3005) [inline vortex-array/src/array/sparse/mod.rs:167]: If you look at previous version of this logic the validity values got flipped here

Comment (a10y) [reply vortex-array/src/array/sparse/mod.rs:167]: wait so this has been wrong for 3 months?

Comment (a10y) [reply vortex-array/src/array/sparse/mod.rs:167]: ...maybe we should add a test? i can also add in a FLUP

Comment (robert3005) [reply vortex-array/src/array/sparse/mod.rs:167]: I can add a test

**Review by a10y — APPROVED** [review]

File: encodings/datetime-parts/src/array.rs [hunk 2]
```diff
@@ -11,7 +11,7 @@ use vortex_error::{vortex_bail, VortexResult};
 
 use crate::compute::decode_to_temporal;
 
-impl_encoding!("vortex.datetimeparts", 20u16, DateTimeParts);
+impl_encoding!("vortex.datetimeparts", 22u16, DateTimeParts);
```
Comment (a10y) [inline encodings/datetime-parts/src/array.rs:14]: oh whoops, this actually collides with runendbools

Comment (robert3005) [reply encodings/datetime-parts/src/array.rs:14]: lol

Comment (robert3005) [reply encodings/datetime-parts/src/array.rs:14]: https://github.com/spiraldb/vortex/pull/498

**Pull request merged**